### ISSUE TITLE
Panel Editor: Support dashboard variable for Max Data Points

### ISSUE THREAD

**Issue by leolchat** [grafana/grafana#28487]
![image](https://user-images.githubusercontent.com/34161/96923438-267edc80-1466-11eb-8fc7-5e70f77f6207.png)

**What would you like to be added:**

I would like to use a dashboard variable for "Max data points". Right now assigning a dashboard variable is silently ignored. If it's not possible it should display an error message. 

I have tried to find in the source code how the field is handled and it seems to call `parseInt` directly, which would explain the behaviour.

It is surprising for the user since "Min interval" works fine with a variable.
And I have really been missing that feature.

**Environment**:
- Grafana version: 7.1.1
- Data source type & version: all data sources are affected

**Comment by ivanahuckova** [comment]
@leolchat first of all, I am really surprised that you were able to add template variable there as the **Max data points** input type is **number**, therefore it is not possible to write text there. But other than that, yes you are right, all the other fields interpolate variables, but **Max data points**  doesn't. So @grafana/grafana-frontend-platform not sure if this is bug or feature request, but passing it to your backlog.

**Comment by leolchat** [comment]
@ivanahuckova my grafana version might not have that *number* constraint, firefox on macos and linux and safari allow me to enter text.

**Comment by jackw** [comment]
Having investigated this, it appears to be an [issue with firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1398528) which allows input fields with `type="number"` attribute to accept non numeric characters. Picking through various react related issues around this problem it seems it can be solved by making use of `e.target.validity.valid` to prevent unwanted characters in number inputs.

**Comment by leolchat** [comment]
@jackw interesting, I did check on safari 14.0 (macos) and saw the same behavior.
I really would like to be able to put a variable there, while it seems that you are leaning toward enforcing the restriction. Should I open another issue for that?

**Comment by jackw** [comment]
Apologies @leolchat for giving the impression that I was leaning toward enforcing the number restriction. I've gone ahead and edited this issue so it's considered a feature request (feel free to update it further if I've missed anything) and I will open a separate one related to the firefox number input issue.

**Comment by Guddr** [comment]
Hi @leolchat, for 2022 it turns out you can't change the values for maxDataPoints through variables?

**Comment by stolsvik** [comment]
We just got hit by this, where we would like to set the "Max data points" using a variable, not a fixed number.

The feature is well described in the OP. As he mentions, "Min interval" does handle variables, so why does not "Max data points" do this? It would be a very nice feature.

**Comment by HeTvaM** [comment]
After asking the question, I have been using "min interval" for a long time, it turned out even better than I expected, trying to limit "max data points"

**Comment by mckn** [comment]
@grafana/dashboards-squad

**Comment by torkelo** [comment]
the maxDataPoints field / option is a number option (schema / model wise) so not would complicate things to support string variable expression there. 

The interval field is better suited for this, you can create an interval variable to control it via variable

**Comment by stolsvik** [comment]
"so not would complicate" was a bit hard to parse? @torkelo 

If it means that it would be hard - why is that?

**Comment by torkelo** [comment]
I mean "so would complicate". 

@stolsvik it is currently a number field / option which does not accept strings, used in many places and passed around to backend and other places

**Comment by stolsvik** [comment]
Can't you just number-parse the interpolated value upon reception or at the processing point? Fail if it contains letters? This seems like a minor technological problem which removes a pretty obvious functionality which does have merit and meaning.

**Comment by mdvictor** [comment]
This feature request has been open for a long time with few received upvotes and it also has a workaround, so we are closing it. We're trying to limit open GitHub issues in order to better track planned work and features.
This doesn't mean that we'll never ever implement it or that we will never accept a PR for it. A closed issue can still attract upvotes and act as a ticket to track feature demand/interest.
Thank you for taking the time to create this issue!
